typings/pinejs-client-core.d.ts

```diff
@@ -252,10 +252,23 @@ export type SubmitBody<T> = {
 	[k in keyof T]?: T[k] extends AssociatedResource ? number | null : T[k];
 };
 
+type BaseResourceId =
+	| string
+	| number
+	| Date
+	| {
+			'@': string;
+	  };
+type ResourceId =
+	| BaseResourceId
+	| {
+			[key: string]: BaseResourceId;
```

Comment:
Should this be:
```suggestion
			[key: keyof T]: BaseResourceId;
```
?

Reply:
done